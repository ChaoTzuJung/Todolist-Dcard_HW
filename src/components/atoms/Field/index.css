.field {
	& > input {
		width: 100%;
		height: 30px;
		padding: 0 12px;
		font-size: 12px;
		font-weight: 300;
		letter-spacing: 0.5px;
		border: 1px solid var(--white);
		outline: none;
		transition: border 0.2s;

		/* &:focus {
			border: 1px solid var(--lightGray);
		} */
	}

	& > .border {
		border: none;
	}

	&.icon {
		display: inline-flex;
		width: 100%;
		height: 35px;

		& > i {
			display: flex;
			align-items: center;
			justify-content: center;
			width: 35px;
			font-size: 18px;
			border: 1px solid var(--gray);
			border-right: 0;
			color: var(--black);
			transition: border 0.2s;
		}

		& > input {
			height: 100%;
		}
	}

	&.file > input {
		border: none;
	}

	&.invalid {
		& > i,
		& > input {
			border-color: #d0021b;
		}
	}
}
