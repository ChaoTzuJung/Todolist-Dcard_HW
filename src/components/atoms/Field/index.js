import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const Field = ({ type, icon, name, className, valid = true, color, border, ...other }) => (
	<label
		htmlFor={name}
		className={classnames(styles.field, className, {
			[styles.file]: type === 'file',
			[styles.icon]: icon !== '',
			[styles.invalid]: !valid,
		})}
	>
		{icon && <i className="material-icons">{icon}</i>}
		<input
			className={classnames(styles.input, {
				[styles.border]: border,
			})}
			style={{ background: color }}
			autoComplete="off"
			type={type}
			name={name}
			{...other}
		/>
	</label>
);

export default Field;
