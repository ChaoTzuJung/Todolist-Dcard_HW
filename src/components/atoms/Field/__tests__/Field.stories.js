import React from 'react';
import { withKnobs, select, text, boolean, color } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import Field from 'components/atoms/Field';

const stories = storiesOf('Commons|atoms/Field', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => (
	<Field
		type={select('type', {
			text: 'text',
			number: 'number',
			password: 'password',
			file: 'file',
			tel: 'tel',
			url: 'url',
		})}
		name="phone"
		icon={text('with icon', 'person_outline')}
		placeholder="aaa"
		valid={boolean('valid', true)}
		color={color('color', '#FFFFFF')}
		border={boolean('border', false)}
	/>
));
