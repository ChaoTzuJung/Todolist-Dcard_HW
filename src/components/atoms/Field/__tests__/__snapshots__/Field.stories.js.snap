// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<div
  style={
    Object {
      "margin": "60px",
    }
  }
>
  <Field
    border={true}
    color="rgba( 0, 0, 0, 0)"
    icon="person_outline"
    name="phone"
    placeholder="type something here"
    valid={true}
  >
    <label
      className="field icon"
      htmlFor="phone"
    >
      <i
        className="material-icons"
      >
        person_outline
      </i>
      <input
        autoComplete="off"
        className="input border"
        name="phone"
        placeholder="type something here"
        style={
          Object {
            "background": "rgba( 0, 0, 0, 0)",
          }
        }
      />
    </label>
  </Field>
</div>
`;
